internal/deploymentcontext/disk_provider.go

```diff
@@ -0,0 +1,195 @@
+package deploymentcontext
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/alecthomas/atomic"
+	"github.com/alecthomas/errors"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/block/ftl/common/log"
+	schemapb "github.com/block/ftl/common/protos/xyz/block/ftl/schema/v1"
+	"github.com/block/ftl/common/schema"
+)
+
+type diskSecretsProvider struct {
+	secretsPath string
+}
+
+type diskConfigProvider struct {
+	configsPath string
+}
+
+type diskRouteProvider struct {
+	schemaPath string
+	schema     atomic.Value[*schema.Schema]
+}
+
+var _ RouteProvider = (*diskRouteProvider)(nil)
+
+func NewDiskSecretsProvider(path string) SecretsProvider {
+	provider := &diskSecretsProvider{
+		secretsPath: path,
+	}
+	return provider.GetSecrets
+}
+
+// GetSecrets reads secrets from individual files in the configured directory
+func (p *diskSecretsProvider) GetSecrets(ctx context.Context) map[string][]byte {
+	logger := log.FromContext(ctx)
+	secrets := make(map[string][]byte)
+
+	entries, err := os.ReadDir(p.secretsPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			logger.Warnf("Secrets directory %s does not exist", p.secretsPath)
+			return secrets
+		}
+		logger.Errorf(err, "Failed to read secrets directory %s", p.secretsPath)
+		return secrets
+	}
+
+	for _, entry := range entries {
+		if entry.IsDir() || strings.HasPrefix(entry.Name(), ".") { // Skip directories and hidden files
+			continue
+		}
+		filePath := filepath.Join(p.secretsPath, entry.Name())
+		fileContentBytes, err := os.ReadFile(filePath)
+		if err != nil {
+			logger.Errorf(err, "Failed to read secret file %s", filePath)
+			continue
+		}
+		// Marshal the file content as a JSON string before storing
+		// This ensures that if the file contains "foo", it's stored as "\"foo\""
+		// so that GetSecret(..., &myString) correctly unmarshals it.
+		jsonEncodedValue, err := json.Marshal(string(fileContentBytes))
```

Comment:
I don't know if this is right. I think FTL would expect JSON encoded values would be stored in the secret/CM.